react/16/1/src/main.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { useState } from 'react';
+import RedBoxGame from "./red_box_game";
+import DisplayScore from "./game_score";
+
+
+const App = () => {
+  const [score, setScore] = useState(0);

```

Comment:
We know that the game can only increase the score by a known amount or lower it (win or lose)

In such a scenario I'd usually define specific functions:

```
function win() { setScore(s => s + 5); }
function lose() { setScore(s => s - 2); }
```

And pass them to the child component. The rational here is to give each component the minimum tools it needs to work properly